src/main/java/frc/robot/vision/sources/VisionSource.java

```diff
@@ -0,0 +1,18 @@
+package frc.robot.vision.sources;
+
+import frc.robot.vision.filters.Filter;
+import frc.robot.vision.rawdata.RawVisionData;
+
+import java.util.Optional;
+
+public interface VisionSource<T extends RawVisionData> {
```

Comment:
consider renaming T to a name that represent what its extends

Reply:
@GilStein1 you reverted that change in the past. @NoamRosenberg08 also suggested it.

Reply:
Yes, it once had a different name, Hillel and Yoni insisted I change it to T